src/controllers/DynamicHierarchicalFacetQueryController.ts

```diff
@@ -6,12 +6,23 @@ import { FacetValueState } from '../rest/Facet/FacetValueState';
 import { IQueryResults } from '../rest/QueryResults';
 import { findIndex } from 'underscore';
 import { IDynamicHierarchicalFacet, IDynamicHierarchicalFacetValue } from '../ui/DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { DynamicFacetRequestBuilder } from './DynamicFacetRequestBuilder';
+import { IQuery } from '../rest/Query';
 
 export class DynamicHierarchicalFacetQueryController {
+  private requestBuilder: DynamicFacetRequestBuilder;
   private numberOfValuesToRequest: number;
   private freezeFacetOrder = false;
 
   constructor(private facet: IDynamicHierarchicalFacet) {
+    this.requestBuilder = new DynamicFacetRequestBuilder({
+      facetId: this.facet.options.id,
+      field: this.facet.fieldName,
+      type: this.facet.facetType,
+      injectionDepth: this.facet.options.injectionDepth,
+      delimitingCharacter: this.facet.options.delimitingCharacter,
+      filterFacetCount: this.facet.options.filterFacetCount
```

Comment:
Related to my previous comment, we can see that all it's doing is just reading options on `this.facet.options` so it would be nice if instead you could pass in either a `DynamicFacet` or `DynamicFacetOptions` kind of interface.